app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt

```diff
@@ -235,12 +239,14 @@ fun ActivityDetailsScreen(
               // price
               Row(
                   verticalAlignment = Alignment.CenterVertically,
-                  modifier = Modifier.testTag("price")) {
+                  modifier = Modifier.testTag("price").fillMaxWidth()) {
                     Icon(Icons.Filled.AttachMoney, contentDescription = "Price")
                     Spacer(modifier = Modifier.width(SMALL_PADDING.dp))
                     Text(
                         text = if (price != null) "${price.toString()} CHF" else "not defined yet",
                         modifier = Modifier.testTag("priceText"))
+                    Spacer(modifier = Modifier.weight(WIDTH_FRACTION))
+                    PaymentInfoScreen(price ?: 0.0)
```

Comment:
Super handling of the null price case. Null price would imply a free activity. This works as we expect.